pkgs/by-name/ma/managarr/package.nix

```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  fetchFromGitHub,
+  gitUpdater,
+  rustPlatform,
+  perl,
+}:
+
+let
+  pname = "managarr";
+  version = "0.4.2";
+in
+
+rustPlatform.buildRustPackage {
+  inherit pname version;
+  src = fetchFromGitHub {
+    owner = "Dark-Alex-17";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-sZDkvIVFCilXNRhoeGudoPE+/62yIMElFoOx+XESWOA=";
+  };
+
+  cargoHash = "sha256-owXHTMf7aI/QHUCls0udzpPXTCqB5urMPvrMBtQFCXM=";
+
+  nativeBuildInputs = [ perl ];
+
+  passthru.updateScript = gitUpdater { rev-prefix = "v"; };
```

Comment:
Does it support updating cargoHash as well?

Reply:
It's a question I have asked myself as well, but looking at generic-updater.nix, probably no.

I'll look into nix-update and commit again when I've looked into it and can confirm it's working.

Thanks for the time and effort

Reply:
Nix-update supports updating cargoHash

Reply:
Thinking about it, I'm not sure there is an easy way to quickly calculate a cargoHash or I'm just not reading correctly. `nix-update` mentions it, but I couldn't make it work locally, it would just update the version number.

Quickly glancing through some Rust packages, I don't see any making use of updateScript, so I have removed it.

Regarding the commit history, is it fine to keep it like this or should commits be squashed? I don't know if there's a consensus in nixpkgs.

Reply:
> Nix-update supports updating cargoHash

OK, thanks. For the current version, I'll leave it as is and include nix-update when the next package update hits.

Reply:
> Regarding the commit history, is it fine to keep it like this or should commits be squashed?

The commits should be squashed, but if someone is unable to do so we can squash merge from GitHub to squash into a single commit. 

> For the current version, I'll leave it as is and include nix-update when the next package update hits

You do you 🤷🏻‍♂️

Reply:
> Thinking about it, I'm not sure there is an easy way to quickly calculate a cargoHash or I'm just not reading correctly. `nix-update` mentions it, but I couldn't make it work locally, it would just update the version number.
> 
> Quickly glancing through some Rust packages, I don't see any making use of updateScript, so I have removed it.
> 
> Regarding the commit history, is it fine to keep it like this or should commits be squashed? I don't know if there's a consensus in nixpkgs.

In my experience, at least in node packages, that use an approach kinda similar, `gitUpdater` updates all these hashes just fine. I'd recommend you start packaging the version just before the last one then when you have something working then test with `maintainers/scripts/update.nix`.